### DISCUSSION TITLE
Stream Deck + dial support

### DISCUSSION THREAD

**Discussion by Atomika** [cgiesche/streamdeck-homeassistant#220] [Help and Support] [answered]
Hi there, sorry to bother.

Would it be at all possible to add support for the rotary dials of the stream deck+? It would be great to be able to control the brightness of a home assistant light entity with the dial (using the rotationPercent or rotationAbsolute?).

At present I believe this is impossible as the only options for lights are toggle, turn on, and turn off. I am a novice though so very possible I am wrong. 

Thank you!

**Comment by kramttocs** [comment]
It already is :)
See Release 2.4.0

**Comment by Atomika** [comment]
Apologies, maybe I'm approaching what I'm trying to do wrong, but I'm now on 2.4.1 and within the 'light' domain, the only services  available are still the 'toggle', 'turn on' and 'turn off' options. I've tried inputting code into the optional JSON data (likely incorrectly, but attempting to link the brightness and rotation variables) but it stays only able to do one of those three things, for example with 'toggle' selected it turns the light on and off with each click of the dial. 

Is a more interactive / incremental dial up and dial down of the brightness possible or is that not supported? Would I need to hook more into things on the homeassistant side to achieve this?

Many thanks and once again apologies for the hassle.

Cheers.

**Comment by cgiesche** [comment, marked as answer]
Hi Atomica,

increasing and decreasing brightness does not work as one may expect in home-assistant. There is no "increase" service, but a parameter for the "turn on" service, that can be used for that:

![image](https://github.com/cgiesche/streamdeck-homeassistant/assets/5123180/985e044a-87e3-450c-9756-90cac7daea95)

You can find available options and variables below the service data field. This is nothing my plugin does but a native feature of home assistant. The `ticks` variable contains the number of ticks the knob was rotated (-x for left turns, +x for right turns).

Official HA-Docs: https://www.home-assistant.io/integrations/light/#service-lightturn_on

**Comment by Atomika** [comment]
That's great, thank you for the help! Thats all great information and has enabled me to do what I was trying to achieve.

Thank you again, much appreciated!
